pyproject.toml

```diff
@@ -388,12 +389,15 @@ lint.ignore = [  # NOTE: non-permanent exclusions should be added to `.ruff.toml
     "B018",  # UselessExpression
     "D",  # pydocstyle
     "PGH001",  # No builtin eval() allowed
+]
+"**/tests/*.py" = [
     "S101",  # Use of assert detected
 ]
 ".pyinstaller/*.py" = ["INP001"]  # Not a package.
 "conftest.py" = ["INP001"]  # Part of configuration, not a package.
 "docs/*.py" = [
     "INP001",  # implicit-namespace-package. The examples are not a package.
+    "S101",  # Use of assert detected
```

Comment:
Here's what the violations in `docs/` look like (in current `main`):
```
$ ruff check --select S101 --show-source docs/
docs/wcs/examples/from_file.py:46:5: S101 Use of `assert` detected
   |
44 |     # These should be the same as the original pixel coordinates, modulo
45 |     # some floating-point error.
46 |     assert np.max(np.abs(pixcrd - pixcrd2)) < 1e-6
   |     ^^^^^^ S101
47 | 
48 |     # The example below illustrates the use of "origin" to convert between
   |

docs/wcs/examples/from_file.py:54:5: S101 Use of `assert` detected
   |
52 |     y = 0
53 |     origin = 0
54 |     assert (w.wcs_pix2world(x, y, origin) ==
   |     ^^^^^^ S101
55 |             w.wcs_pix2world(x + 1, y + 1, origin + 1))
   |

docs/wcs/examples/programmatic.py:41:1: S101 Use of `assert` detected
   |
39 | # These should be the same as the original pixel coordinates, modulo
40 | # some floating-point error.
41 | assert np.max(np.abs(pixcrd - pixcrd2)) < 1e-6
   | ^^^^^^ S101
42 | 
43 | # The example below illustrates the use of "origin" to convert between
   |

docs/wcs/examples/programmatic.py:49:1: S101 Use of `assert` detected
   |
47 | y = 0
48 | origin = 0
49 | assert (w.wcs_pix2world(x, y, origin) ==
   | ^^^^^^ S101
50 |         w.wcs_pix2world(x + 1, y + 1, origin + 1))
   |

Found 4 errors.
```
We might want to replace these `assert` statements with [`np.testing.assert_allclose()`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html) and [`np.testing.assert_array_equal()`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html#numpy.testing.assert_array_equal) instead of ignoring S101 in `docs/`.

Reply:
That's an option, though I must admit I don't quite like how the checkers here are affecting good & clear code. I somewhat prefer to leave the warts with the checker set-up files rather than with the code, i.e., either just exclude docs -- where presumably the author, like here, made a conscious choice to use `assert` -- or just exclude these two files.

p.s. This PR is to support running python with `-OO` - for that purpose, none of this should really matter...

Reply:
As far as I can tell `docs/wcs/examples/from_file.py` exists only so that `docs/wcs/loading_from_fits.rst` can include it. But the latter contains only 8 lines in total. Perhaps it would be best to move the contents of `from_file.py` to `loading_from_fits.rst`. Likewise for `docs/wcs/examples/programmatic.py` and `docs/wcs/example_create_imaging.rst`.

Reply:
For now, I will move the exclude tule for docs to `ruff.toml` (just to make the CI happy) and maybe a separate PR can be made if the example files need to be modified or moved.

Reply:
That's reasonable.